samples/matter/common/src/dfu_over_smp.cpp

```diff
@@ -134,8 +121,15 @@ void DFUOverSMP::StartBLEAdvertising()
 	}
 }
 
-void DFUOverSMP::OnBleDisconnect(struct bt_conn *conId, uint8_t reason)
+void DFUOverSMP::OnBleConnect(bt_conn *conn, uint8_t err)
```

Comment:
Looks like the operations on QSPI are now handled asynchronously. Is it guaranteed that the flash is ready on time, i.e. before the actual DFU starts?

Reply:
Yes, because we use the system work queue which is also used for SMP command processing.